components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -1118,6 +1120,93 @@ public OAuthRevocationResponseDTO revokeAuthzForAppsByResourceOwner(
+
+        triggerPreApplicationTokenRevokeListeners(application);
+
+        String consumerKey = application.getConsumerKey();
+        String applicationName = application.getApplicationName();
+        if (consumerKey != null && applicationName != null) {
+            List<AccessTokenDO> activeDetailedTokens;
+            try {
+                //Get active access tokens by application id
+                activeDetailedTokens = new ArrayList<>(OAuthTokenPersistenceFactory
+                        .getInstance().getAccessTokenDAO().getActiveAcessTokenDataByConsumerKey(consumerKey));
+            } catch (IdentityOAuth2Exception e) {
+                String errorMsg = "Error occurred while retrieving access tokens issued for Client ID : " + consumerKey;
+                throw handleError(errorMsg, e);
+            }
+            if (activeDetailedTokens.size() > 0) {
+                String[] accessTokens = new String[activeDetailedTokens.size()];
+                int countToken = 0;
+                AuthenticatedUser authzUser;
+                for (AccessTokenDO detailToken : activeDetailedTokens) {
+                    String token = detailToken.getAccessToken();
+                    accessTokens[countToken++] = token;
+
+                    OAuthCacheKey cacheKeyToken = new OAuthCacheKey(token);
+                    String scope = buildScopeString(detailToken.getScope());
+                    String accessToken = detailToken.getAccessToken();
+
+                    //Clear cache with AccessTokenDO
+                    authzUser = detailToken.getAuthzUser();
+                    TokenBinding tokenBinding = detailToken.getTokenBinding();
+                    String tokenBindingReference = (tokenBinding != null &&
+                            StringUtils.isNotBlank(tokenBinding.getBindingReference())) ?
+                            tokenBinding.getBindingReference() : NONE;
+
+                    OAuthCache.getInstance().clearCacheEntry(cacheKeyToken);
+                    OAuthUtil.clearOAuthCache(detailToken.getConsumerKey(), authzUser, scope, tokenBindingReference);
+                    OAuthUtil.clearOAuthCache(detailToken.getConsumerKey(), authzUser, scope);
+                    OAuthUtil.clearOAuthCache(detailToken.getConsumerKey(), authzUser);
+                    OAuthUtil.clearOAuthCache(accessToken);
+                }
+
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Access tokens and token of users are removed from the cache for OAuth App with " +
+                            "consumerKey: " + consumerKey);
+                }
+
+                //Revoking token from database
+                try {
+                    OAuthTokenPersistenceFactory.getInstance().getAccessTokenDAO().revokeAccessTokens(accessTokens);
+                } catch (IdentityOAuth2Exception e) {
+                    String errorMsg = "Error occurred while revoking access Tokens for OAuth App with consumerKey : "
```

Comment:
```suggestion
                    String errorMsg = "Error occurred while revoking access tokens for OAuth App with consumerKey : "
```

Reply:
Addressed in [40fde89](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1627/commits/40fde89cb099493c8f2a302b34b1141d9aea3e8d)